File: src/test/java/com/automation/um/pages/components/ConfirmDialog.java
package com.automation.um.pages.components;

import com.automation.um.pages.BasePage;
import org.openqa.selenium.WebDriver;

public class ConfirmDialog extends BasePage {
  public ConfirmDialog(WebDriver driver) {
    super(driver);
  }

  public void confirm() {
    el("confirm-ok").click();
  }

  public void cancel() {
    el("confirm-cancel").click();
  }

  public boolean isOpen() {
    return isVisible("confirm-ok");
  }
}

File: playwright/pages/components/ConfirmDialog.js
import { BasePage } from "../BasePage.js";

export class ConfirmDialog extends BasePage {
  async confirm() {
    await this.testId("confirm-ok").click();
  }

  async cancel() {
    await this.testId("confirm-cancel").click();
  }

  async isOpen() {
    return this.isVisible("confirm-ok");
  }
}
